### PR TITLE
Test form rendering.

### PR DESCRIPTION
**Pull request by andorp**
* New make command: `test-repl`
 * `FormPageView` has an associated type and a new method to define forms
 * Test suite, to check if the form renders and uses the same parameters as it was given in the view

### CODE REVIEW

File: tests/Frontend/HtmlSpec.hs [hunk 1]
```diff
@@ -59,5 +68,19 @@ data HtmlGen where
...
+    it (show $ typeOf g) . property . forAll g $ \pageSource -> LT.length (renderText (toHtml pageSource)) > 0
+
+data FormGen where
+    F :: (Show m, Typeable m, FormPageView m) => Gen m -> FormGen
+
+-- | Checks if the form rendering does not contains buttons and
+-- the view has all the fields defined for GET form creation.
+renderForm (F g) =
+    it (show $ typeOf g) . property . forAll g $ \page -> monadicIO $ do
+        length <- run . fmap failOnError . runExceptT $ do
```
Comment (np) [inline tests/Frontend/HtmlSpec.hs]: I'd rather avoid using `length` as a variable name.

**Comment by np** [comment]
apart from the comment it LGTM

File: tests/Frontend/HtmlSpec.hs (hunk 1 above)
Comment (andorp) [inline tests/Frontend/HtmlSpec.hs]: I am going to chage it.

File: tests/Frontend/HtmlSpec.hs [hunk 2]
```diff
@@ -59,5 +68,19 @@ data HtmlGen where
...
 -- | Checks if the markup rendering does not contains bottoms.
-renderMarkup (H g) = describe (show $ typeOf g) .
-    it "renders" . property . forAll g $ \pageSource -> LT.length (renderText (toHtml pageSource)) > 0
+renderMarkup (H g) =
+    it (show $ typeOf g) . property . forAll g $ \pageSource -> LT.length (renderText (toHtml pageSource)) > 0
+
+data FormGen where
+    F :: (Show m, Typeable m, FormPageView m) => Gen m -> FormGen
+
+-- | Checks if the form rendering does not contains buttons and
```
Comment (Mikolaj) [inline tests/Frontend/HtmlSpec.hs]: buttons or bottoms?

Comment (andorp) [inline tests/Frontend/HtmlSpec.hs]: Good point. Bottoms. Fixing...

**Comment by Mikolaj** [comment]
Does it subsume #20 (for forms?) Anyway, it's more readable than the old version and the extra test doesn't hurt either. LGTM>

**Pull request merged**